multichain-testing/test/xcs-swap-anything/helpers.ts

```diff
@@ -0,0 +1,826 @@
+    if (txRes && txRes.code !== 0) {
+      console.error(txRes);
+      throw Error(`failed to ibc transfer funds to ${destinationAddress}`);
+    }
+    t.is(txRes.code, 0, `Transaction succeeded`);
+    console.log('sendIbcTokens TxResponse: ', txRes);
+
+    const denom = await getDenomHash(
+      destinationChain,
+      issuingChain,
+      issuingDenom,
+    );
+
+    const getDenomAmount = balances => {
+      const balance = balances.find(coin => coin.denom === `ibc/${denom}`);
+      const amount = balance ? Number(balance.amount) : 0;
+      return amount;
+    };
+
+    await retryUntilCondition(
+      () => destinationClient.getAllBalances(destinationAddress),
+      (coins: Coin[]) => {
+        const before = getDenomAmount(destinationBalanceBefore);
+        const current = getDenomAmount(coins);
+        return current > before;
+      },
+      `Transferred tokens not found on destination address: ${destinationAddress}`,
+    );
+    console.log(
+      `Transferred tokens found on destination address: ${destinationAddress}`,
+    );
+
+    return {
+      client: destinationClient,
+      address: destinationAddress,
+      wallet: destinationWallet,
+    };
+  };
+
+  const invokeOsmosisContract = async (contractAddress: string, txMsg) => {
+    const clientRegistry = osmosisClient.registry;
+    clientRegistry.register(MsgExecuteContract.typeUrl, MsgExecuteContract);
+
+    const fee = {
+      amount: [{ denom: 'uosmo', amount: '1000' }],
+      gas: '200000',
+    };
+
+    const message = MsgExecuteContract.fromPartial({
+      sender: osmosisAddress,
+      contract: contractAddress,
+      msg: toUtf8(JSON.stringify(txMsg)),
+      funds: [],
+    });
+
+    const encodeObjects: Array<import('@cosmjs/proto-signing').EncodeObject> = [
+      {
+        typeUrl: '/cosmwasm.wasm.v1.MsgExecuteContract',
+        value: message,
+      },
+    ];
+
+    const response = await osmosisClient.signAndBroadcast(
+      osmosisAddress,
+      encodeObjects,
+      fee,
+    );
+    console.log('invokeOsmosisContract DeliverTxResponse: ', response);
+
+    const { msgResponses, code } = response;
+
+    if (code !== 0) {
+      throw Error(`Failed to execute osmosis contract with message ${message}`);
+    }
+
+    return msgResponses;
+  };
+
+  const queryOsmosisContract = async (contractAddress: string, queryMsg) => {
+    const wasmClient = await CosmWasmClient.connect(osmosisRpcEndpoint);
+    const result = await wasmClient.queryContractSmart(
+      contractAddress,
+      queryMsg,
+    );
+
+    return result;
+  };
+
+  const downloadXcsContracts = async (
+    xcsContracts: Contracts,
+    branch: string = osmosisBranch,
+    scriptPath: string = scriptLocalPath,
+    artifactsPath: string = xcsArtifactsPath,
+  ) => {
+    const wasmFiles: string[] = [];
+    for (const contract in xcsContracts) {
+      if (Object.prototype.hasOwnProperty.call(xcsContracts, contract)) {
```

Comment:
```suggestion
    for (const contract of xcsContracts) {
```

wow... do our lint rules really allow `for ... in ..`???

Reply:
The lint rules did not raise an issue with the use of for in.
Change request introduced in commit:
cf37ab29da835de9d57537314def04088878ab92

Reply:
for ref, as we discussed...

[guard-for-in](https://eslint.org/docs/latest/rules/guard-for-in#rule-details) does allow `for ... in` provided it has the own property check.

But `for ... of` is more concise.

The only reason I can see to use `for ... in` is in some context where `for ... of` is not available.